### PR TITLE
Now working with Mujoco 1.31 (the latest)

### PR DESCRIPTION
**Pull request by eak24**
To get the gps code base to work with mujoco 1.31, I referred to the [official mujoco changelist.](http://www.mujoco.org/changelist.html)

### CODE REVIEW

**Comment by eak24** [comment]
[Here is a video of the mjc_example experiment](https://youtu.be/hyCwpyupWF4) training using MuJoCo Pro 1.31. The other mjc experiments also work with this PR.

**Comment by eak24** [comment]
@cbfinn I uploaded a video (referenced in the link above) of the mjc_example experiment running with mjpro 1.31. Please let me know if there is anything else you need before accepting the PR!

**Comment by svlevine** [comment]
I looked at the changes, I think everything is fine as far as I can tell. One concern I have is that this will break backward compatibility for anyone still using the older version of MuJoCo. Maybe this is OK and they should upgrade anyway, but it's perhaps a little unfortunate. I made some comments about how this could be rectified, but I'm not sure about one of the files.

EDIT: Actually, maybe we should just have the current users who are using the old MuJoCo version test out this PR to make sure it doesn't break everything for them, it would be cleaner to switch to the new version completely. As far as I know, at UCB current users who might have an issue are: Bradly, Coline, Abhishek, Aviv. At UW, I don't have any issues.

File: src/3rdparty/mjcpy2/CMakeLists.txt [hunk 1]
```diff
@@ -46,4 +47,4 @@ message("osg includes: ${OSG_INCLUDE_DIR}")
 include_directories(SYSTEM "${CMAKE_SOURCE_DIR}")
 include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/Boost.NumPy")
 boost_python_module(mjcpy mjcpy2.cpp  mujoco_osg_viewer.cpp)
-target_link_libraries(mjcpy "${CMAKE_SOURCE_DIR}/mjpro/libmujoco.so" ${OSG_LIBRARIES} boost_numpy)
+target_link_libraries(mjcpy "${CMAKE_SOURCE_DIR}/mjpro/bin/libmujoco131.so" ${OSG_LIBRARIES} boost_numpy)
```
Comment (svlevine) [inline src/3rdparty/mjcpy2/CMakeLists.txt:50]: Can we add support for cmake flag to use the old version? Then we can phase it out once everyone switches over. Probably fine to default to the new version.

File: src/3rdparty/mjcpy2/mjcpy2_getmodel_autogen.i [hunk 2]
```diff
@@ -156,7 +156,7 @@
     out["tex_rgb"] = toNdarray2<mjtByte>(m_model->tex_rgb, m_model->ntexdata, 1);
     out["mat_texid"] = toNdarray2<int>(m_model->mat_texid, m_model->nmat, 1);
     out["mat_texuniform"] = toNdarray2<mjtByte>(m_model->mat_texuniform, m_model->nmat, 1);
-    out["mat_texscale"] = toNdarray2<float>(m_model->mat_texscale, m_model->nmat, 2);
+    out["mat_texrepeat"] = toNdarray2<float>(m_model->mat_texrepeat, m_model->nmat, 2);
```
Comment (svlevine) [inline src/3rdparty/mjcpy2/mjcpy2_getmodel_autogen.i:159]: I guess mat_texrepeat is not used somewhere? I would expect it to be used in the renderer, but I guess not. Could probably just remove.

File: src/3rdparty/mjcpy2/mjcpy2_setmodel_autogen.i [hunk 3]
```diff
@@ -156,7 +156,7 @@
     _cadihk(d, "tex_rgb", m_model->tex_rgb);
     _cadihk(d, "mat_texid", m_model->mat_texid);
     _cadihk(d, "mat_texuniform", m_model->mat_texuniform);
-    _cadihk(d, "mat_texscale", m_model->mat_texscale);
+    _cadihk(d, "mat_texrepeat", m_model->mat_texrepeat);
```
Comment (svlevine) [inline src/3rdparty/mjcpy2/mjcpy2_setmodel_autogen.i:159]: As above, if texrepeat is not used, could remove for backward compatibility.

File: src/3rdparty/mjcpy2/mujoco_osg_viewer.cpp [hunk 4]
```diff
@@ -478,7 +478,7 @@ void MujocoOSGViewer::SetData(const mjData* d) {
 
 void NewModelFromXML(const char* filename,mjModel*& model) {
     char errmsg[100];
-    model = mj_loadXML(filename, errmsg);
+    model = mj_loadXML(filename, NULL, errmsg, sizeof(errmsg));
```
Comment (svlevine) [inline src/3rdparty/mjcpy2/mujoco_osg_viewer.cpp:481]: Not sure what to do about this one. If we end up adding that cmake flag, could use it here to choose the appropriate function signature.

**Comment by eak24** [comment]
Thanks for the recommendations. I don't have access to the old mjpro library, so I cannot verify that any changes work. Do you have the ability to verify backwards compatibility?

**Comment by cbfinn** [comment]
Sorry for the terribly slow response on this. [I'm more available to work on this now, than I was before, so you can expect much faster response times going forward.]

I just tested this PR on a computer with an old version of mujoco on got the following compile error:
/home/cfinn/code/gps_public/gps/src/3rdparty/mjcpy2/mujoco_osg_viewer.cpp: In function ‘void NewModelFromXML(const char_, mjModel_&)’:
/home/cfinn/code/gps_public/gps/src/3rdparty/mjcpy2/mujoco_osg_viewer.cpp:481:62: error: too many arguments to function ‘mjModel\* mj_loadXML(const char_, char_)’
In file included from /home/cfinn/code/gps_public/gps/src/3rdparty/mjcpy2/mujoco_osg_viewer.hpp:5:0,
                 from /home/cfinn/code/gps_public/gps/src/3rdparty/mjcpy2/mujoco_osg_viewer.cpp:2:
/home/cfinn/code/gps_public/gps/src/3rdparty/mjpro/mujoco.h:90:16: note: declared here
make[2]: **\* [mjcpy2/CMakeFiles/mjcpy.dir/mujoco_osg_viewer.cpp.o] Error 1
In file included from /home/cfinn/code/gps_public/gps/src/3rdparty/mjcpy2/mjcpy2.cpp:279:0:
/home/cfinn/code/gps_public/gps/src/3rdparty/mjcpy2/mjcpy2_getmodel_autogen.i: In member function ‘boost::python::dict {anonymous}::PyMJCWorld2::GetModel()’:
/home/cfinn/code/gps_public/gps/src/3rdparty/mjcpy2/mjcpy2_getmodel_autogen.i:159:55: error: ‘mjModel’ has no member named ‘mat_texrepeat’
In file included from /home/cfinn/code/gps_public/gps/src/3rdparty/mjcpy2/mjcpy2.cpp:283:0:
/home/cfinn/code/gps_public/gps/src/3rdparty/mjcpy2/mjcpy2_setmodel_autogen.i: In member function ‘void {anonymous}::PyMJCWorld2::SetModel(boost::python::dict)’:
/home/cfinn/code/gps_public/gps/src/3rdparty/mjcpy2/mjcpy2_setmodel_autogen.i:159:42: error: ‘mjModel’ has no member named ‘mat_texrepeat’
make[2]: **\* [mjcpy2/CMakeFiles/mjcpy.dir/mjcpy2.cpp.o] Error 1
make[1]: **\* [mjcpy2/CMakeFiles/mjcpy.dir/all] Error 2
make: **\* [all] Error 2

File: src/3rdparty/mjcpy2/CMakeLists.txt (hunk 1 above)
Comment (cbfinn) [inline src/3rdparty/mjcpy2/CMakeLists.txt:50]: Something else to consider here is that the mjpro directory is now called mjpro131 on download, so we should either change this to be mjpro131/ or make this clear in the install instructions to rename the directory.

**Comment by cbfinn** [comment]
I went ahead and fixed backwards compatibility issues in https://github.com/cbfinn/gps/pull/40. Thanks for this PR!

**Pull request merged**